src/idl_parser.cpp

```diff
@@ -897,6 +897,29 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
     EXPECT(',');
   }
 
+  // Check if all required fields are parsed.
+  for (auto field_it = struct_def.fields.vec.begin();
+            field_it != struct_def.fields.vec.end();
+            ++field_it) {
+    auto required_field = *field_it;
+    if (!required_field->required) {
+      continue;
+    }
+    bool found = false;
+    for (auto pf_it = field_stack_.begin();
```

Comment:
It probably won't find it anymore now that you are comparing field objects, which are different. It should still happen if you compare names instead. Also, it can happen with a recursive table, like `table A { b:B (required); a:A; }`. Here, it could pick up a missing `b` from an out object, since the field pointers will actually be equal.